src/commons/crypto/signing/dispatch/krillsigner.rs

```diff
@@ -217,30 +222,30 @@ impl KrillSigner {
         Serial::random(&self.router).map_err(crypto::Error::signer)
     }
 
-    pub fn sign<D: AsRef<[u8]> + ?Sized>(&self, key_id: &KeyIdentifier, data: &D) -> CryptoResult<Signature> {
+    pub fn sign<D: AsRef<[u8]> + ?Sized>(&self, key_id: &KeyIdentifier, data: &D) -> CryptoResult<RpkiSignature> {
         self.router
-            .sign(key_id, SignatureAlgorithm::default(), data)
+            .sign(key_id, RpkiSignatureAlgorithm::default(), data)
             .map_err(crypto::Error::signing)
     }
 
-    pub fn sign_one_off<D: AsRef<[u8]> + ?Sized>(&self, data: &D) -> CryptoResult<(Signature, PublicKey)> {
+    pub fn sign_one_off<D: AsRef<[u8]> + ?Sized>(&self, data: &D) -> CryptoResult<(RpkiSignature, PublicKey)> {
         self.router
-            .sign_one_off(SignatureAlgorithm::default(), data)
+            .sign_one_off(RpkiSignatureAlgorithm::default(), data)
             .map_err(crypto::Error::signer)
     }
 
-    pub fn sign_csr(&self, base_repo: &RepoInfo, name_space: &str, key: &KeyIdentifier) -> CryptoResult<Csr> {
+    pub fn sign_csr(&self, base_repo: &RepoInfo, name_space: &str, key: &KeyIdentifier) -> CryptoResult<RpkiCaCsr> {
```

Comment:
It is clear enough to me, but if it is not clear enough to others then that is a sign.

I will take it up with @partim - a better solution might be to return a `RpkiCaCsr` in rpki-rs, rather than having to reparse it. We could also use the builder and separate sign function pattern that we have used elsewhere, but in this case - because the number of arguments is quite limited my preference would be not to do that. @partim thoughts?

Reply:
I think changing `Csr::construct_rpki_csr` to return a `Csr` rather than just the bytes is a good idea. This is certainly cheaper than making and then decoding it.

Reply:
ok, then let me merge this first. I created issue #843 so that we can get back to this later. There are number of PRs on top of each other in both krill and rpki-rs and doing this in between would complicate things.